macros/src/main/scala/MacroCompiler.scala

```diff
@@ -92,16 +99,9 @@ object MacroCompilerAnnotation {
    * @param c Top-level circuit name (see class description)
    * @param p Parameters (see above).
    */
-  def apply(c: String, p: Params): Annotation =
-    Annotation(CircuitName(c), classOf[MacroCompilerTransform], MacroCompilerUtil.objToString(p))
+  def apply(c: String, p: Params): MacroCompilerAnnotation =
+    MacroCompilerAnnotation(MacroCompilerUtil.objToString(p))
```

Comment:
If anyone is wondering why the factory method pattern is useful, this is why. We were able to change the implementation of this method and it's actually binary compatible (not that binary compatibility really matters in this case).